### DISCUSSION TITLE
Changing docker services credentials in .env for self host budibase with docker-compose

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#10050] [Help] [answered]
Hi devs,

I wanted to confirm with you that if I change any of the following security values values in .env I will face no future issues, provided I do a docker compose down/up -d. I already tested this and Budibase seams to happily accept all changes in env vars and apps continue to work as before.  I also tested cli backup/restore to a Budibase host with different var values and all worked fine I think. However I would like your confirmation that these vars are not connected in anyway with apps operation.

```
# This section contains all secrets pertaining to the system
# These should be updated
API_ENCRYPTION_KEY=testsecret
JWT_SECRET=testsecret
MINIO_ACCESS_KEY=budibase
MINIO_SECRET_KEY=budibase
COUCH_DB_PASSWORD=budibase
COUCH_DB_USER=budibase
REDIS_PASSWORD=budibase
INTERNAL_API_KEY=budibase

```
Thank you!

**Comment by melohagan** [comment, marked as answer]
Hey @gmentzik 

The values of the environment variables will be inserted into your apps at the point of startup (when you do a docker-compose up). 

Therefore this will only impact your app when initialising. Changing these values for an already running app will have no effect.

**Reply by gmentzik** [reply]
Hi @melohagan I appreciate the response. 
Ok so if an application in budibase is already created with some creds then if you alter env vars and do docker-compose down/up it will fail. Already verified that. Importing it again as new seems to work though, but this would be expected.

What I need is a confirmation about the migration procedure using cli to get apps to a new Budibase host with different env credentials than the original. My test showed that apps were working properly using cli to backup at host1 with env1 and cli to restore to host2 with env2. Can this procedure be trusted or even in the case or cli backup and restore the recommendation is to have exactly the same .env var values?

**Reply by gmentzik** [reply]
Hi @melohagan. Could you please confirm if it ok to use cli to backup and restore between two Budibase self hosts with different .env parameter values? It is not clear from documentation.

**Reply by shogunpurple** [reply]
@gmentzik It is generally recommended to use the same environment variables on a brand new budibase instance across the two servers just to avoid any issues. Values such as the `API_ENCRYPTION_KEY` should be maintained between the 2 instances or can result in backups being corrupted.
